src/content/getusermedia/resolution/js/main.js

```diff
@@ -52,17 +59,62 @@ var fullHdConstraints = {
 function gotStream(mediaStream) {
   window.stream = mediaStream; // stream available to console
   video.srcObject = mediaStream;
+  messagebox.style.display = 'none';
+  videoblock.style.display = 'block';
+  let track = mediaStream.getVideoTracks()[0];
+  let constraints = track.getConstraints();
+  console.log('Result constraints: ' + JSON.stringify(constraints));
+  if (constraints && constraints.width && constraints.width.exact) {
+    widthInput.value = constraints.width.exact;
+    widthOutput.textContent = constraints.width.exact;
+  } else if (constraints && constraints.width && constraints.width.min) {
+    widthInput.value = constraints.width.min;
+    widthOutput.textContent = constraints.width.min;
+  }
+}
+
+function errorMessage(who, what) {
+  let message = who + ': ' + what;
+  messagebox.innerHTML = message;
+  messagebox.style.display = 'block';
+  console.log(message);
+}
+
+function clearErrorMessage() {
+  messagebox.style.display = 'none';
 }
 
 function displayVideoDimensions() {
-  if (!video.videoWidth) {
-    setTimeout(displayVideoDimensions, 500);
-  }
-  dimensions.innerHTML = 'Actual video dimensions: ' + video.videoWidth +
+  if (video.videoWidth) {
+    dimensions.innerHTML = 'Actual video dimensions: ' + video.videoWidth +
     'x' + video.videoHeight + 'px.';
+    console.log(dimensions.innerHTML);
+  } else {
+    dimensions.innerHTML = 'Video not ready';
+  }
 }
 
 video.onloadedmetadata = displayVideoDimensions;
+// Video size takes a few seconds to change when size changes,
+// so we track it every half second.
+setInterval(displayVideoDimensions, 500);
```

Comment:
actually (and thanks @pehrsons) this is https://bugs.chromium.org/p/chromium/issues/detail?id=601470 you're working around here. Maybe fix the actual bug? :-)